src/lavinmq/clustering/actions.cr

```diff
@@ -14,6 +14,7 @@ module LavinMQ
 
       abstract def lag_size : Int64
       abstract def send(socket : IO, log = Log) : Int64
+      abstract def abort
```

Comment:
What about naming it `done` and call it from the action loop when an `Action` has been sent, instead of assuming that `send` will be called only once for an `Action` and call `unreserve` from inside `send`?

Reply:
makes sense, thanks!

Reply:
put `done` in `ensure` in the `Action#send` method instead

Reply:
yes, the assumption is that `Action#send` is only called once

Reply:
Instead of assuming that, which would be an internal implementation detail the caller may need to know about, I think it would make sense to do 

```crystal
action.send(lz4, Log)
action.done
```

in the action loop.

Reply:
It gets ugly because you have to have a begin/recue block, and then you have two loops, so two blocks on top of that

Reply:
True